rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,625 @@
+* Security: @jentfoo
+
+## What
+
+Connect My Computer (further abbreviated as CMC) enables the user of Teleport Connect to add the
+device running Teleport Connect to the cluster as an SSH node.
+
+The MVP achieves that by downloading the teleport binary, creating the relevant role and the join
+token and then starting the agent from within the Electron app. Since the device is added as a
+regular node, this poses a security concern of enabling any cluster user with sufficient permissions
+to log into the device running Connect My Computer. This and other concerns are addressed in [the
+Security section](#security).
+
+The goal of this RFD is to gather feedback on the core of Connect My Computer, that is managing the
+node agent running on the same device as Connect. Decisions on how to drive the users toward this
+feature, such as adding a Discover tile or deep linking from the Web UI to Connect, are not covered
+in this RFD and will be covered in future RFDs.
+
+The wireframes included in the RFD are supposed to convey the general idea behind the UI. The setup
+tab and the status tab are likely to stay in the same format. The top bar UI and the CTAs are likely
+to change, although they are likely to receive no extra new features. If you have feedback on the UI
+which goes beyond the general flow and towards the specifics, please leave it directly on [the
+document in Figma](https://www.figma.com/file/jeKtOZrFsIVOtKH7yd8Xiy/Connect-My-Computer-UI?type=design).
+
+## Why
+
+The idea behind Connect My Computer is to reduce [the time to
+magic](https://beneinstein.medium.com/min-time-to-magic-c938d9ea3473). By enabling the user to add a
+new node to the cluster with just a few clicks, we hope to get them to the wow moment as quickly as
+possible. We want to show that configuring and accessing the infrastructure with Teleport is fast
+and easy. This will be further amplified by adding new features further down the line, such as
+sharing sessions.
+
+## Details
+
+After signing into a Team plan cluster, which currently has no connected nodes, the user sees a
+prompt in the server resource table encouraging them to use CMC (screenshot 1 below). Interacting
+with the CTA opens a new tab which explains what happens during the setup and lets the user start
+the setup (screenshot 2 below).
+
+![CTA](assets/0133-cta.png)
+
+![setup landing](assets/0133-setup-landing.png)
+
+The setup requires no extra configuration. It sets up relevant roles, reissues the user cert to
+refresh the roles, downloads the teleport binary, creates a join token, calls `teleport node
+configure`, and then launches the agent by calling `teleport start`.
+
+![setup progress](assets/0133-setup-progress.png)
+
+
+The setup steps are displayed in a similar fashion to Cloud account setup steps when the cluster is
+being spun up for the first time. Once the setup is done, the setup tab automatically transitions to
+showing the status of the agent.
```

Comment:
Will Teleport Connect be usable while this setup is happening, or will the user be effectively blocked at a loading screen?

Reply:
> Will Teleport Connect be usable while this setup is happening, or will the user be effectively blocked at a loading screen?

The app will be usable. In theory this means you can switch to another workspace and set up Connect My Computer there at the same time too. The MVP will simply ensure that the second setup won't corrupt the first one.